### PR TITLE
Move master_update_shard_statistics() to pg_catalog

### LINKED ISSUE #546: master_update_shard_statistics function sits in wrong schema
Should be in pg_catalog:

    postgres=# \df
                                           List of functions
     Schema |              Name              | Result data type |   Argument data types   |  Type  
    --------+--------------------------------+------------------+-------------------------+--------
     public | master_update_shard_statistics | bigint           | shard_id bigint         | normal

### PR DESCRIPTION
**Pull request by metdos**
Fixes #546

### CODE REVIEW

File: src/backend/distributed/citus--5.1-2--5.1-3.sql [hunk 1]
```diff
@@ -0,0 +1,8 @@
+DROP FUNCTION master_update_shard_statistics(bigint);
```
Comment (marcocitus) [inline src/backend/distributed/citus--5.1-2--5.1-3.sql]: An alternative is to just ALTER FUNCTION .. SET SCHEMA.

**Comment by marcocitus** [comment]
Random unrelated question in relation to #522 : Does master_update_shard_statistics work with tables in a schema?

**Comment by marcocitus** [comment]
Interesting, the travis output says that master_update_shard_statistics is already in pg_catalog when doing the alter extension. There seems to be a difference between the create extension and the alter extension, which suggests a search path issue in one of the SQL files. Anyway, seems like dropping and re-creating might actually be the better approach then, because at this point some people might have it in pg_catalog while others have it in public. We should also figure out what's going on there.

**Comment by metdos** [comment]
> Interesting, the travis output says that master_update_shard_statistics is already in pg_catalog when doing the alter extension. There seems to be a difference between the create extension and the alter extension, which suggests a search path issue in one of the SQL files. Anyway, seems like dropping and re-creating might actually be the better approach then, because at this point some people might have it in pg_catalog while others have it in public. We should also figure out what's going on there.

I noticed this yesterday but didn't have time to check it. In the previous "create function", there were not any specific schema, so ALTER EXTENSION creates it in pg_catalog, but if you create extension newly, the function is created in the public schema.

I switched back to drop and create.

**Comment by metdos** [comment]
> Random unrelated question in relation to #522 : Does master_update_shard_statistics work with tables in a schema?

Yes, we indirectly test this in COPY append with schemas. I also tested it separately.

**Comment by marcocitus** [comment]
Tested it. Looks good to me 🚢 .

**Comment by jasonmp85** [comment]
Is this something we want to fix in the release (i.e. 5.1) or are we fine leaving this for 5.2?

**Comment by metdos** [comment]
> Is this something we want to fix in the release (i.e. 5.1) or are we fine leaving this for 5.2?

I think that we are fine with leaving this for 5.2. @marcocitus, any objections?

**Comment by marcocitus** [comment]
I think leaving it for 5.2 is fine.

**Pull request merged**